components/code_block/code_block.tsx

```diff
@@ -75,7 +75,20 @@ const CodeBlock: React.FC<Props> = ({id, code, language, searchedContent}: Props
         htmlContent = `${searchedContent} ${content}`;
     }
 
+    const [codeBlockSelection, setCodeBlockSelection] = useState('');
+
+    const setSelection = () => {
+        const selectedText = document.getSelection();
+        if (selectedText) {
+            setCodeBlockSelection(selectedText.toString());
+        }
+    };
```

Comment:
this can lead to bugs, since selecting something and then lifting that selection does not clear the state and copying would still just copy the text from the last selection.

See below for a different approach.

Reply:
Commented on the approach above, but in theory this fn is only (and always) called when the contextMenu is opened/onShow (with right-click), and is only used for the contextMenu copy. So under normal use case it shouldn't be possible to copy an old selection, since trying to access the copy option necessarily resets the selection. Please correct me if I missed something though!

Reply:
as stated on the other comment this is ok to be merged as is.